### PR TITLE
rgw: test that no asio threads are blocked on null_yield

### PR DESCRIPTION
**Pull request by cbodley**
continuation of the async refactoring project to eliminate the use of null_yield in frontend request coroutines

* moves the existing warnings into new function `maybe_warn_about_blocking()` in rgw_asio_thread.*
* adds config option `rgw_asio_assert_yielding` that teuthology tests can enable to crash on assertion failure instead of logging a warning message
* adds `maybe_warn_about_blocking()` coverage to some other blocking functions

as we make progress eliminating these null_yields, we can enable `rgw_asio_assert_yielding` in teuthology jobs to catch any regressions

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
when enabling `rgw_asio_assert_yielding` in a vstart cluster, i was able to do simple operations like multipart uploads and bucket creation/listing/deletion. but running s3-tests against that still crashes:

> FAILED s3tests_boto3/functional/test_s3.py::test_lifecycle_set - botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:8000/yournamehere-lojqimpnwzlbkdts-361?lifecycle"

this at least gives us a process to locate and fix any such crashes, then eventually enable `rgw_asio_assert_yielding` in teuthology to catch any regressions

**Comment by cbodley** [comment]
from https://jenkins.ceph.com/job/ceph-api/76357/:
> FAIL: test_standby_disable_redirect (tasks.mgr.test_dashboard.TestDashboard)

commented on https://tracker.ceph.com/issues/45528

**Comment by cbodley** [comment]
jenkins test api

File: src/common/options/rgw.yaml.in [hunk 1]
```diff
@@ -2230,6 +2230,14 @@ options:
   services:
```
Comment (adamemerson) [inline src/common/options/rgw.yaml.in:2230]: Nit: The first line of this commit is too long

**Review by adamemerson — APPROVED** [review]

**Comment by cbodley** [comment]
from https://jenkins.ceph.com/job/ceph-api/77046/
> teuthology.exceptions.CommandFailedError: Command failed with status 250: '../src/vstart.sh -n --nolockdep'

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2024-07-03_01:53:26-rgw-wip-cbodley-testing-distro-default-smithi/ with rerun https://pulpito.ceph.com/cbodley-2024-07-04_21:53:21-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**